### PR TITLE
Sync with Refract Spec

### PR DESCRIPTION
**Pull request by smizell**
This handle issue #34.

This is kind of... nasty. But it aligns the code and terminology used through with the spec. This breaks every possible interface for the module :)

cc @danielgtaylor

### CODE REVIEW

**Comment by zdne** [comment]
:boom: :boom: :boom: :boom:   :dancers:

File: README.md [hunk 1]
```diff
@@ -36,12 +36,12 @@ var el = {
 
 ## Usage
 
-### Converting to Types
+### Converting to Elements
```
Comment (danielgtaylor) [inline README.md]: I feel like maybe this is more clear if we mention both the `to` and `from`:

Converting Javascript Values into Elements

File: README.md [hunk 2]
```diff
@@ -74,12 +74,12 @@ The `refract` variable above has the following JSON value.
 }
 ```
 
-### Converting from Types
+### Converting from Elements
```
Comment (danielgtaylor) [inline README.md]: Converting Refract JSON into Elements
Converting Serialized Refract into Elements

File: README.md (hunk 1 above)
Comment (smizell) [inline README.md]: I think you are 100% correct

File: README.md [hunk 3]
```diff
@@ -74,12 +74,12 @@ The `refract` variable above has the following JSON value.
...
 
-If the JSON above is used, it can be converted back to Minim types to make a roundtrip.
+If the JSON above is used, it can be converted back to Minim elements to make a roundtrip.
 
 ```javascript
-var arrayType = minim.convertFromDom(aboveJson);
+var arrayElement = minim.convertFromElement(aboveJson);
```
Comment (danielgtaylor) [inline README.md]: In this method the word `element` seems to refer to a JSON serialization of refract, while above `convertToElement` uses the word `element` to mean an instance of an element class. I wonder if we can't make this a little more clear / less ambiguous.

``` js
var arrayElement1 = minim.fromValue([1, 2, 3]);
var refracted = arrayElement1.toRefract();
var arrayElement2 = minim.fromRefract(refracted);
```

Edit: the individual element classes use `toElement` to convert from values, so maybe that part makes sense. I don't know - trying to think of this from a newcomer checking out the library.

File: lib/base.js [hunk 4]
```diff
@@ -87,22 +87,22 @@ var TypeRegistry = createClass({
...
   },
 
   /*
-   * Convert a long-form refract DOM into refract type instances.
+   * Convert a long-form refract document into refract element instances.
    */
-   fromRefract: function(dom) {
-    var ElementClass = this.getElementClass(dom.element);
-    return new ElementClass().fromRefract(dom);
+   fromRefract: function(doc) {
```
Comment (danielgtaylor) [inline lib/base.js:99]: :+1: doc over dom

**Comment by danielgtaylor** [comment]
Generally I agree with this change, just a few minor points above!

**Comment by smizell** [comment]
@danielgtaylor I've addressed comments. I'd like to squash some of these before merging, so let me know if it's good and I'll merge. Thanks!

**Comment by danielgtaylor** [comment]
Other than that comment above I think this looks good. We'll need to take some time to update the downstream libraries like Fury.

**Comment by smizell** [comment]
> We'll need to take some time to update the downstream libraries like Fury

I'm going to be working on this a lot over the next few days. I'm wondering if it makes sense to wait until I'm finish (which will be the end of our sprint next Tuesday).

**Comment by smizell** [comment]
Also, I sent out a document about some proposed changes for the API Description namespace. I think it would be good to wait on that, too, and fix it all at once.

**Pull request merged**